src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -131,7 +131,7 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   protected def findEntry(key: A): Entry =
     findEntry0(key, index(elemHashCode(key)))
 
-  private[this] def findEntry0(key: A, h: Int): Entry = {
+  protected[this] final def findEntry0(key: A, h: Int): Entry = {
```

Comment:
My fix isn't about the inlining limit, that's a side-effect of collapsing the stack depth by calling the indexing methods earlier.

This fix would speed up 2.11 also.

Please advise on how you want me to change this PR.
I could start by adding a JMH benchmark for `HashMap` and verifying the performance of other methods also - and trying to optimize them on the way.
Are we sure we want to do the benchmarks from `Scala` and not `Java`?